Mathlib/Analysis/Normed/Group/Quotient.lean

```diff
@@ -93,59 +93,206 @@ the previous paragraph kicks in.
+
+/-- The norm of the image of `m : M` in the quotient by a closed subgroup `S` is zero if and only if
+`m ∈ S`. -/
+@[to_additive "The norm of the image of `m : M` in the quotient by a closed subgroup `S` is zero if
+and only if `m ∈ S`."]
+lemma norm_mk'_eq_zero [hS : IsClosed (S : Set M)] : ‖mk' S m‖ = 0 ↔ m ∈ S := norm_mk_eq_zero
+
+/-- For any `x : M ⧸ S` and any `0 < ε`, there is `m : M` such that `mk' S m = x`
+and `‖m‖ < ‖x‖ + ε`. -/
+@[to_additive "For any `x : M ⧸ S` and any `0 < ε`, there is `m : M` such that `mk' S m = x`
+and `‖m‖ < ‖x‖ + ε`."]
+lemma exists_norm_mk'_lt (x : M ⧸ S) (hε : 0 < ε) : ∃ m : M, mk' S m = x ∧ ‖m‖ < ‖x‖ + ε :=
+  norm_lt_iff.1 <| lt_add_of_pos_right _ hε
+
+/-- For any `m : M` and any `0 < ε`, there is `s ∈ S` such that `‖m * s‖ < ‖mk' S m‖ + ε`. -/
+@[to_additive
+"For any `m : M` and any `0 < ε`, there is `s ∈ S` such that `‖m + s‖ < ‖mk' S m‖ + ε`."]
+lemma exists_norm_mul_lt (S : Subgroup M) (m : M) {ε : ℝ} (hε : 0 < ε) :
+    ∃ s ∈ S, ‖m * s‖ < ‖mk' S m‖ + ε := by
+  obtain ⟨n : M, hn : mk' S n = mk' S m, hn' : ‖n‖ < ‖mk' S m‖ + ε⟩ :=
+    exists_norm_mk'_lt (QuotientGroup.mk' S m) hε
+  exact ⟨m⁻¹ * n, by simpa [eq_comm, QuotientGroup.eq] using hn, by simpa⟩
+
+variable (S) in
+/-- The seminormed group structure on the quotient by a subgroup. -/
+@[to_additive "The seminormed group structure on the quotient by an additive subgroup."]
+noncomputable instance instSeminormedCommGroup : SeminormedCommGroup (M ⧸ S) where
+  toUniformSpace := IsTopologicalGroup.toUniformSpace (M ⧸ S)
+  __ := groupSeminorm.toSeminormedCommGroup
+  uniformity_dist := by
+    rw [uniformity_eq_comap_nhds_one', (nhds_one_hasBasis.comap _).eq_biInf]
+    simp only [dist, preimage_setOf_eq, norm_eq_groupSeminorm, map_div_rev]
+
+variable (S) in
+/-- The quotient in the category of normed groups. -/
+@[to_additive "The quotient in the category of normed groups."]
+noncomputable instance instNormedCommGroup [hS : IsClosed (S : Set M)] :
+    NormedCommGroup (M ⧸ S) where
+  __ := MetricSpace.ofT0PseudoMetricSpace _
+
+-- This is a sanity check left here on purpose to ensure that potential refactors won't destroy
+-- this important property.
+example :
+    (instTopologicalSpaceQuotient : TopologicalSpace <| M ⧸ S) =
+      (instSeminormedCommGroup S).toUniformSpace.toTopologicalSpace := rfl
+
+example [IsClosed (S : Set M)] :
+   (instSeminormedCommGroup S) = NormedCommGroup.toSeminormedCommGroup := rfl
+
+end QuotientGroup
+
 open QuotientAddGroup Metric Set Topology NNReal
 
 variable {M N : Type*} [SeminormedAddCommGroup M] [SeminormedAddCommGroup N]
 
 /-- The definition of the norm on the quotient by an additive subgroup. -/
-noncomputable instance normOnQuotient (S : AddSubgroup M) : Norm (M ⧸ S) where
+@[deprecated QuotientAddGroup.instNorm (since := "2025-02-02")]
+noncomputable def normOnQuotient (S : AddSubgroup M) : Norm (M ⧸ S) where
   norm x := sInf (norm '' { m | mk' S m = x })
```

Comment:
I think you should replace this with `inferInstance`, I think leaving this definition is going to cause more problems not fewer.